File: ios/Mirror/libs/TerraWallet.m
//
//  TerraWallet.m
//  Harvest
//
//  Created by Felix on 2020/04/03.
//

//method export를 위함.

#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(TerraWallet, NSObject)

+ (BOOL)requiresMainQueueSetup
{
  return false;  // only do this if your module initialization relies on calling UIKit!
}

RCT_EXTERN_METHOD(isValidAddress:(NSString *)address
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/Mirror/libs/TerraWallet.swift
import Foundation
/*
 bridging header에
 #import <TrezorCrypto/TrezorCrypto.h>
 #import <React/RCTBridgeModule.h>
 추가할것.
*/

@objc(TerraWallet)
class TerraWallet: NSObject {
  
  
    static func moduleName() -> String! {
        return "TerraWallet";
    }

    static func requiresMainQueueSetup() -> Bool {
        //If your module does not require access to UIKit, then you should respond to + requiresMainQueueSetup with NO.
        return false
    }
  
    @objc func isValidAddress(_ address:String,
                              resolver: RCTPromiseResolveBlock,
                              rejecter:RCTPromiseRejectBlock) {
        let result = TerraWalletSDK.isValidAddress(address);
        resolver(result);
    }
}
